src/SymEngine.jl

```diff
@@ -149,3 +149,14 @@ show(io::IO, b::Basic) = print(io, toString(b))
 
 end
 
+abstract Basic
+
```

Comment:
@isuruf I have gone through a number of discussions on inheritance in Julia like [https://groups.google.com/forum/#!topic/julia-dev/eA4VkFAD-yQ] , [https://groups.google.com/forum/#!msg/julia-dev/p9bV6rMSmvY/cExeSb-cMBYJ]

Julia is not a supporter of inheritance as we have in C++ and Python
we can think of Julia as more a Method Oriented language that an Object Oriented one

For classes like Number and Integer (of Symengine), we have rewrite all the methods for Basic.
As I was going through few julia libraries like [gtk.jl](https://github.com/JuliaLang/Gtk.jl/blob/master/src/gtktypes.jl) , I found the extensive use of macros to do away with rewriting of methods

Problem with heirarchy is that 
abstract A

type B <: A
....
end
 
now we can't have 
type C <: B
...
end

The SymEngine Hierarchy looks difficult to be implemented in Julia.
@isuruf I would like to have your views. 
Now I am having a look at macros that how it can ease of things for us